GH-39910: [Go] Add func to load prepared statement from ActionCreatePreparedStatementResult

abandy opened the pull request:
Currently, in order to create a PreparedStatement a DoAction call will always be made via the client. I need to be able to make a PreparedStatement from persisted data that will not trigger the DoAction call to the server.
* Closes: #39910

zeroshade commented on go/arrow/flight/flightsql/client.go:
```diff
@@ -450,6 +450,31 @@ func (c *Client) PrepareSubstrait(ctx context.Context, plan SubstraitPlan, opts
 	return parsePreparedStatementResponse(c, c.Alloc, stream)
 }
 
+func (c *Client) LoadPreparedStatementFromResult(result *pb.ActionCreatePreparedStatementResult) (*PreparedStatement, error) {
```
rather than exposing the `pb.ActionCreatePreparedStatementResult` directly, please add a type for it in `types.go` with the other protobuf aliasing we do instead. We try to avoid using the protobuf types *directly* in the interface

abandy commented:
Updated the Load prepared statement to use a type defined in types.go and added a func that allows the prepared statement to execute a put without returning a flightinfo.

zeroshade commented:
Thanks @abandy!
